checks/prometheus_check.py

```diff
@@ -118,26 +126,56 @@ def parse_metric_family(self, buf, content_type):
                     else:
                         self.log.debug("type override %s for %s is not a valid type name" % (new_type, message.name))
                 yield message
-        elif 'text/plain' in content_type:
-            messages = {}  # map with the name of the element (before the labels) and the list of occurrences with labels and values
-            obj_map = {}   # map of the types of each metrics
-            obj_help = {}  # help for the metrics
-            for line in buf.splitlines():
-                self._extract_metrics_from_string(line, messages, obj_map, obj_help)
 
-            # Add type overrides:
-            for m_name, m_type in self.type_overrides.iteritems():
-                if m_type in self.METRIC_TYPES:
-                    obj_map[m_name] = m_type
-                else:
-                    self.log.debug("type override %s for %s is not a valid type name" % (m_type,m_name))
+        elif 'text/plain' in response.headers['Content-Type']:
+            messages = defaultdict(list)  # map with the name of the element (before the labels)
+            # and the list of occurrences with labels and values
+
+            obj_map = {}  # map of the types of each metrics
+            obj_help = {}  # help for the metrics
+            for metric in text_fd_to_metric_families(response.iter_lines(chunk_size=1024*10)):
```

Comment:
10kB seems OK for the chuck size, could you move it to a module constant + add comment so we don't have to do to the `requests` apidoc to know the unit and compute the value?